Sources/SwiftLayout/Core/Activator/Component/WeakConstraint.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  WeakConstraint.swift
+//  
+//
+//  Created by oozoofrog on 2022/02/12.
+//
+
+import UIKit
+
+struct WeakConstraint {
+    private(set) weak var origin: NSLayoutConstraint?
+    private var customHashValue: Int
+
+    init(origin: NSLayoutConstraint? = nil) {
+        self.origin = origin
+
+        var hasher = Hasher()
+        hasher.combine(origin?.firstItem as? NSObject)
+        hasher.combine(origin?.firstAttribute)
+        hasher.combine(origin?.secondItem as? NSObject)
+        hasher.combine(origin?.secondAttribute)
+        hasher.combine(origin?.relation)
+        hasher.combine(origin?.constant)
+        hasher.combine(origin?.multiplier)
+        hasher.combine(origin?.priority)
+        self.customHashValue = hasher.finalize()
+    }
+}
+
+extension WeakConstraint: Hashable, Equatable {
+    func hash(into hasher: inout Hasher) {
+        hasher.combine(origin.map( { _ in customHashValue }))
+    }
+
+    static func == (lhs: WeakConstraint, rhs: WeakConstraint) -> Bool {
+        lhs.hashValue == rhs.hashValue
+    }
+}
+
+extension Collection where Element: NSLayoutConstraint {
+    var weakens: [WeakConstraint] {
+        map(WeakConstraint.init)
+    }
+}
```

Comment:
@gmlwhdtjd how about removing this? anytime at seeing this, feeling original sin.

Reply:
I currently have some uses, what do you want to do?

Reply:
use map(WeakConstraint.init)

😄

Reply:
oh it just ask your opinion. and not for this pr this time.